text/0000-bundle-add.md

```diff
@@ -0,0 +1,132 @@
+## 1. add a single gem
+#### Proposed command
+```ruby
+bundle add "GEM"
+
+#examples
+bundle add "rails"
+```
+#### Description
+This is the base use case. The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM with it.
+2. If it fails to resolve correctly or is an invalid GEM, it will error out.
+3. If it resolves successfully, it will install the latest version of the GEM.
+4. It will add the GEM to the Gemfile with a comment and place a conservative version update requirement e.g.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM", "~> 1.0.0"
+```
+
+## 2. add a single gem with options
+#### Proposed command
+```ruby
+# First priority
+bundle add "GEM" [--version VERSION]
+
+# Possible full implementation
+bundle add "GEM" [--version VERSION] [--source SOURCE] [--group GROUP]
+
+#examples
+bundle add "rails" --version "~> 5.0.0"
+bundle add "rails" --version ">= 5.0, < 5.1"
+bundle add "rails" --version "~> 5.0.0" --group "development, test"
+bundle add "rails" --version "~> 5.0.0" --source "https://gems.example.com" --group "development"
+```
+#### Description
+This extends the base case allowing a user to specify optional version, source, and group.
+
+The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM with it.
+2. If it fails to resolve correctly or is an invalid GEM/options, it will error out.
+3. If it resolves successfully, it will install the specified or latest version of the GEM.
+4. It will add the GEM to the Gemfile with a comment and the specified options. If the version is ">=0", then it will substitute it with a conservative version update requirement.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM" --version "~> 5.0.0"
+  gem "GEM" --version ">= 5.0", "< 5.1"
+  gem "GEM" --version "~> 5.0.0", group: ["development", "test"]
+  gem "GEM" --version "~> 5.0.0", source: "https://gems.example.com", group: "development"
+```
+## 3. add multiple gems
+#### Proposed command
+```ruby
+bundle add "GEM1", "GEM2", "GEM3"
+
+#examples
+bundle add "rails", "devise", "simple_form"
+```
+#### Description
+This extends the base case allowing multiple gems to be added.
+The following steps will occur when the command is executed:
+1. It will load the current Gemfile and resolve the new GEM's with it.
+2. If it fails to resolve correctly or there are invalid GEM's, it will error out.
+3. If it resolves successfully, it will install the latest version of each GEM's.
+4. It will add the GEM's to the Gemfile and place a conservative version update requirement e.g.
+```ruby
+  # Gemfile
+  # Added at 2016-12-07 00:21:12 -0800 by username:
+  gem "GEM1", "~> 1.0.0"
+  gem "GEM2", "~> 1.0.0"
+  gem "GEM3", "~> 1.0.0"
+```
+
+## 4. add multiple gems with options
+#### Proposed command
+```ruby
+# TBD
+bundle add "MULTIPLE_GEMS + OPTIONS ???"
```

Comment:
I think supporting options and multiple gems at the same time is probably too much. We can punt on this option and not plan to implement it for now.